### PR TITLE
yaml: phase 1

### PR DESCRIPTION
**Pull request by sougou**
This implements part of #6068.

`db` and `init` still need to be done. TabletConfig has been refactored to match the new specs, and a new flag `tablet_config` has been added to vttablet. This flag is currently experimental and subject to change.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -157,11 +157,13 @@ require (
 	gopkg.in/mgo.v2 v2.0.0-20160818020120-3f83fa500528 // indirect
 	gopkg.in/ory-am/dockertest.v3 v3.3.4 // indirect
 	gopkg.in/square/go-jose.v2 v2.3.1 // indirect
+	gopkg.in/yaml.v2 v2.2.8
```
Comment (harshit-gangal) [inline go.mod:160]: is this used?

Comment (sougou) [reply go.mod:160]: It got auto-added, but didn't get auto-removed.

File: go/vt/vttablet/tabletserver/tabletenv/config.go [hunk 2]
```diff
@@ -43,81 +53,119 @@ var (
...
+	flag.BoolVar(&enableConsolidator, "enable-consolidator", true, "This option enables the query consolidator.")
+	flag.BoolVar(&enableConsolidatorReplicas, "enable-consolidator-replicas", false, "This option enables the query consolidator only on replicas.")
+	flag.BoolVar(&currentConfig.CacheResultFields, "enable-query-plan-field-caching", defaultConfig.CacheResultFields, "This option fetches & caches fields (columns) when storing query plans")
 }
 
 // Init must be called after flag.Parse, and before doing any other operations.
 func Init() {
+	// IdleTimeout is only initialized for OltpReadPool , but the other pools need to inherit the value.
+	currentConfig.OlapReadPool.IdleTimeoutSeconds = currentConfig.OltpReadPool.IdleTimeoutSeconds
+	currentConfig.TxPool.IdleTimeoutSeconds = currentConfig.OltpReadPool.IdleTimeoutSeconds
```
[the remark this answers is not in the document]
Comment (sougou) [reply go/vt/vttablet/tabletserver/tabletenv/config.go:144]: I don't know yet. This code just keeps it backward compatible. We should make a decision on it before making it official.

File: go/vt/vttablet/tabletserver/connpool/pool.go [hunk 3]
```diff
@@ -52,20 +53,24 @@ type Pool struct {
...
+	waiterCap          int64
+	waiterCount        sync2.AtomicInt64
 	dbaPool            *dbconnpool.ConnectionPool
 	appDebugParams     dbconfigs.Connector
 }
 
 // New creates a new Pool. The name is used
 // to publish stats only.
-func New(env tabletenv.Env, name string, capacity int, prefillParallelism int, timeout, idleTimeout time.Duration) *Pool {
+func New(env tabletenv.Env, name string, cfg tabletenv.ConnPoolConfig) *Pool {
```
[the remark this answers is not in the document]
Comment (sougou) [reply go/vt/vttablet/tabletserver/connpool/pool.go:64]: This is a go convention: we should avoid repeating the name of the package within contents of the package: https://blog.golang.org/package-names.

File: go/vt/vttablet/tabletserver/schema/engine.go [hunk 4]
```diff
@@ -67,13 +67,15 @@ type Engine struct {
...
-	idleTimeout := time.Duration(env.Config().IdleTimeout * 1e9)
+	reloadTime := time.Duration(env.Config().SchemaReloadIntervalSeconds * 1e9)
 	se := &Engine{
 		env: env,
 		// We need only one connection because the reloader is
 		// the only one that needs this.
-		conns:      connpool.New(env, "", 1, 0, 0, idleTimeout),
+		conns: connpool.New(env, "", tabletenv.ConnPoolConfig{
+			Size:               1,
+			IdleTimeoutSeconds: env.Config().OltpReadPool.IdleTimeoutSeconds,
```
[the remark this answers is not in the document]
Comment (sougou) [reply go/vt/vttablet/tabletserver/schema/engine.go:77]: This is just backward compatible behavior. We need to make a decision.

**Comment by sougou** [comment]
> 1. Few flags are transformed to yaml config, rest would continue to be used as-is?
This is only a start. If we like this, we should look at transitioning all flags to yaml.

> 2. If flag is provided and also it is present in yaml config. Is yaml config authoritative?
Yaml is authoritative. It will overwrite command line flags. We want to deprecate command line flags.

> 3. Is there any new field added in yaml config which does not have a flag?
Not in this PR, but we'll soon be adding new elements that won't have a command line equivalents.

File: go/vt/vttablet/tabletserver/connpool/pool.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (harshit-gangal) [reply go/vt/vttablet/tabletserver/connpool/pool.go:64]: Package name is connpool. 
The struct name is Pool. This is still allowed as per the convention :)

**Pull request merged**